labgrid/external/remote_tmpdir.py

```diff
@@ -0,0 +1,53 @@
+"""
+    RemoteTmpdir is a class that maintains a temporary directory on a
+    remote linux system.
+    Ex. when copying a test script or binary to DUT, RemoteTmpdir assists
+    to prevent pollution of the target.
+
+    RemoteTmpdir is to be created in a fixture, which can take care of a
+    testmodule in a subdirectory. Recommended scope is function, to avoid
+    polution between tests.
+
+    See examples/remote_tmpdir
+"""
+
+import os
+
+
+class RemoteTmpdir:
+
+    # scriptdir is a directory relative to pytest root
+    def __init__(self, shell, testdir=None):
```

Comment:
There is no rule that a CommandProtocol must support FileTransferProtocol, so I suggest to use
```python
def __init__(self, shell, filetransfer=None, basedir=None):
```

and then later in `__init__()`

```python
if filetransfer is None:
    filetransfer = shell
self.filetransfer = filetransfer
self.basedir = basedir
```

The `basedir` argument can then be used to support relative local paths.